### ISSUE TITLE
What happens if texts from the dataset don't have equal lengths

### ISSUE THREAD

**Issue by sentialx** [lucidrains/recurrent-memory-transformer-pytorch#6]
What would happen if the texts in the dataset didn't have equal lengths and batch size was > 1. E.g. first text would have 2 segments and second 4. Would the loss be nan for segments with padding tokens only? I think your training script assumes that a segment is always of sequence length of 2048 non-padded.

**Comment by lucidrains** [comment]
@sentialx hey Eryk! that should be handled if you pad your sequences with the `ignore_index`, which i have set to `-1` but is configurable

https://github.com/lucidrains/recurrent-memory-transformer-pytorch/blob/main/recurrent_memory_transformer_pytorch/recurrent_memory_transformer.py#LL432C13-L432C25

**Comment by lucidrains** [comment]
@sentialx if everything in the second segment batch were the ignore index (zero in your case), it would be `nan`, but because there is `[1, 2, 0, 0]` you are still ok

**Comment by sentialx** [comment]
@lucidrains one more question, did you have a chance to compare it with block recurrent?
